opentelemetry-sdk/src/opentelemetry/sdk/_metrics/instrument.py

```diff
@@ -30,64 +28,41 @@
 from opentelemetry._metrics.instrument import (
     ObservableUpDownCounter as APIObservableUpDownCounter,
 )
-from opentelemetry._metrics.instrument import TCallback
 from opentelemetry._metrics.instrument import UpDownCounter as APIUpDownCounter
+from opentelemetry.sdk._metrics.aggregation import (
+    ExplicitBucketHistogramAggregation,
+    LastValueAggregation,
+    SumAggregation,
+)
+from opentelemetry.sdk._metrics.measurement import Measurement
 from opentelemetry.sdk.util.instrumentation import InstrumentationInfo
 
+TCallback = Union[
+    Callable[[], Iterable[Measurement]],
+    Generator[Iterable[Measurement], None, None],
+]
 
-class _Instrument:
-    def __init__(
-        self,
-        instrumentation_info: InstrumentationInfo,
-        name: str,
-        *args,
-        unit: str = "",
-        description: str = "",
-    ):
-        self._instrumentation_info = instrumentation_info
-        super().__init__(name, *args, unit=unit, description=description)
 
-
-class Counter(_Instrument, APICounter):
-    def __init__(
-        self,
-        instrumentation_info: InstrumentationInfo,
-        name: str,
-        unit: str = "",
-        description: str = "",
-    ):
-        super().__init__(
-            instrumentation_info,
-            name,
-            unit=unit,
-            description=description,
-        )
-
-    def add(self, amount, attributes=None):
-        # FIXME check that the amount is non negative
+class _Instrument(ABC):
+    @property
+    @abstractmethod
+    def _default_aggregation(self):
```

Comment:
I'd prefer to not have this on the instrument class because 

1. it's "private" to hide it from the user, but then the SDK relies on the private method anyway (and pylint will warn)
2. There is a use case for changing the SDK's default aggregation for all instances of a given instrument. The main use case I heard was for histogram instruments to either change the default explicit histogram buckets or to change all histogram to use the [ExponentialHistogram](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#exponentialhistogram) aggregation (once available) as the default.

Reply:
The default aggregation is an intrinsic property of the instrument, it is better to have it directly coupled with the instrument. Having it elsewhere means that when in need to know the default aggregation, an object has to know about some other object that holds this information instead of having to know only about the instrument. Also, if a new instrument is added, this other object needs to be updated as well.

I have made `default_aggregation` a public attribute, it is mentioned in the spec, so it is not an implementation detail of our SDK. Later something new can make it necessary to add another mechanism to update the default aggregation of an instrument, it is still better to keep the default aggregation in the instrument for the reasons above because they will still apply even if the default aggregation has been changed somehow.

Reply:
I think it will be confusing for users to see this `default_aggregation` as a public property of the instrument. It shouldn't be part of the instrument's public API imo.

Reply:
I don't think it would be confusing because it is clearly stated in the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#default-aggregation) and this just represents that pretty straightforwardly. This represents the association between instrument class and aggregation class as it is defined in the table linked above, this is why we are using a `property` instead of having a method that instantiates the aggregation.

Reply:
I can see how knowing the default aggregation for an instrument can be handy but why does it return class type?

Reply:
It's just representing the relationship stated [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#default-aggregation) which is between an instrument class and an aggregation class as well.

Reply:
I am trying look at this from an end user usability perspective. The relationship alone isn't much helpful for me. I would have expected this maybe returns aggregator (with defaults) to use. What are we trying to accomplish with this?

Reply:
Ok, I have removed it for now :v: